parts/k8s/cloud-init/masternodecustomdata.yml

```diff
@@ -530,6 +530,79 @@ MASTER_CONTAINER_ADDONS_PLACEHOLDER
     {{WrapAsVariable "environmentJSON"}}
 {{end}}
 
+{{- if IsGuardAddonEnabled}}
+- path: /etc/kubernetes/guard/pki/ca.crt
+  permissions: "0644"
+  encoding: "base64"
+  owner: root
+  content: |
+    {{GetGuardCertificate "guardCACertificate"}}
+
+- path: /etc/kubernetes/guard/pki/azure@azure.crt
+  permissions: "0644"
+  encoding: "base64"
+  owner: root
+  content: |
+    {{GetGuardCertificate "guardClientCertificate"}}
+
+- path: /etc/kubernetes/guard/pki/azure@azure.key
+  permissions: "0644"
+  encoding: "base64"
+  owner: root
+  content: |
+    {{GetGuardCertificate "guardClientCertificateKey"}}
+
+- path: /etc/kubernetes/guard/guard-authn-webhook.yaml
+  permissions: "0644"
+  owner: root
+  content: |
+    apiVersion: v1
+    clusters:
+    - cluster:
+        certificate-authority: /etc/kubernetes/guard/pki/ca.crt
+        server: https://guard.kube-system.svc.cluster.local:443/tokenreviews
+      name: guard-server
+    contexts:
+    - context:
+        cluster: guard-server
+        user: azure@azure
+      name: webhook
+    current-context: webhook
+    kind: Config
+    preferences: {}
+    users:
+    - name: azure@azure
+      user:
+        client-certificate: /etc/kubernetes/guard/pki/azure@azure.crt
+        client-key: /etc/kubernetes/guard/pki/azure@azure.key
+#EOF
```

Comment:
The purpose of these sentinel "#EOF" characters is to be used with the CSE helper func `wait_for_file`. Is there CSE-specific logic that will be implemented that depends upon this file being present at the time that the CSE shell commands are executed?

Reply:
Not at the moment. However apiserver requires this file to be present by the time it starts to boot up provided webhook authentication is set. It will crash otherwise.

Reply:
In that case we should probably add something like this before the kublet systemd service:

```
{{- IsGuardAddonEnabled}}
    wait_for_file 1200 1 /etc/kubernetes/guard/guard-authn-webhook.yaml || exit {{GetCSEErrorCode "ERR_FILE_WATCH_TIMEOUT"}}
{{- end}}
```

(we want a `wait_for_file` for every critical file that kubelet systemd will fail when starting up)